File: services/helidon-mp-svc/src/main/java/com/perfsim/helidonmpsvc/domain/Order.java
package com.perfsim.helidonmpsvc.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.OffsetDateTime;

@Entity
@Table(name = "orders", schema = "helidon_mp")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    @Column(nullable = false)
    public String customer;

    @Column(nullable = false, length = 32)
    public String status = "PENDING";

    @Column(name = "total_cents", nullable = false)
    public long totalCents;

    @Column(name = "created_at", nullable = false, insertable = false, updatable = false)
    public OffsetDateTime createdAt;
}

File: services/dotnet-svc/Models/Order.cs
namespace DotnetSvc.Models;

public sealed class Order
{
    public long Id { get; set; }
    public string Customer { get; set; } = "";
    public string Status { get; set; } = "PENDING";
    public long TotalCents { get; set; }
    public DateTime CreatedAt { get; set; }
}
